ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -681,6 +694,139 @@ public void shouldNotIncludeSslPropertiesInListPropertiesOutput() {
         not(hasItems(KsqlConfig.SSL_CONFIG_NAMES.toArray(new String[0]))));
   }
 
+  @Test
+  public void shouldFailForInvalidTerminateClusterParameters() {
+    // Given:
+    final Map<String, Object> properties = new HashMap<>();
+    properties.put("DELETE_TOPIC_LIS", Collections.singletonList("FOO"));
+    final KsqlRequest ksqlRequest = new KsqlRequest(TerminateCluster.TERMINATE_CLUSTER_STATEMENT_TEXT, properties);
+    final KsqlResource testResource = TestKsqlResourceUtil.get(ksqlConfig, ksqlEngine);
+
+    // When:
+    final Response response = testResource.terminateCluster(ksqlRequest);
+
+    // Then:
+    assertThat(response.getStatus(), equalTo(500));
+    assertThat(response.getEntity().toString(), startsWith("Invalid request parameter"));
+  }
+
+  private static class TestKsqlResourceUtil {
```

Comment:
`TestKsqlResourceUtil` is pretty weird... 

Have a go at refactoring
- make use of mocks, rather than things like `TestCommandConsumer`
- have the mocks as fields
- setup common behaviour in the `setUp` method
- create the thing under test in `setUp`, i.e. KsqlResource, passing in mocks.